modules/core/src/main/java/org/apache/ignite/internal/client/thin/io/gridnioserver/GridNioClientConnectionMultiplexer.java

```diff
@@ -146,8 +146,30 @@ public GridNioClientConnectionMultiplexer(ClientConfiguration cfg) {
         rwLock.readLock().lock();
 
         try {
-            SocketChannel ch = SocketChannel.open();
-            ch.socket().connect(new InetSocketAddress(addr.getHostName(), addr.getPort()), connTimeout);
+            SocketChannel ch = null;
+            try {
+                ch = SocketChannel.open();
+                ch.socket().connect(new InetSocketAddress(addr.getHostName(), addr.getPort()), connTimeout);
+            }
+            catch (Exception e) {
+                if (ch != null) {
+                    if (ch.socket() != null) {
+                        try {
+                            ch.socket().close();
+                        }
+                        catch (Exception ex) {
+                            throw new ClientConnectionException(ex.getMessage(), ex);
+                        }
+                    }
+                    try {
+                        ch.close();
+                    }
+                    catch (Exception ex) {
+                        throw new ClientConnectionException(ex.getMessage(), ex);
+                    }
+                }
```

Comment:
```suggestion
                if (ch != null) {
                    if (ch.socket() != null) {
                        try {
                            ch.socket().close();
                        }
                        catch (Exception ignored) {
                        }
                    }
                    
                    try {
                        ch.close();
                    }
                    catch (Exception ignored) {
                    }
                }
```

The idea is to throw the original exception to the user. We don't care about exceptions that happen int `close()`.

Reply:
I agree. Can we do some log here?